homeassistant/components/rympro/__init__.py

```diff
@@ -0,0 +1,82 @@
+"""The Read Your Meter Pro integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from pyrympro import CannotConnectError, OperationError, RymPro, UnauthorizedError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD, CONF_TOKEN, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+SCAN_INTERVAL = 60 * 60
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Read Your Meter Pro from a config entry."""
+    data = entry.data
+    rympro = RymPro(async_get_clientsession(hass))
+    rympro.set_token(data[CONF_TOKEN])
+    try:
+        await rympro.account_info()
+    except CannotConnectError as error:
+        raise ConfigEntryNotReady from error
+    except UnauthorizedError:
+        try:
+            token = await rympro.login(data[CONF_EMAIL], data[CONF_PASSWORD], "ha")
+            hass.config_entries.async_update_entry(
```

Comment:
Please only wrap the line that can raise in the try... except block. Move this line down.